### PR TITLE
[Bug] Make sure table generation works on timestamp columns with timezone information

### PR DESCRIPTION
**Pull request by yiweny**
fix upto function on data with timezone information and add test cases

### CODE REVIEW

**Review by weihua916 — COMMENTED** [review]
What is this testing? No assert statement. It'd be good to add an expected behavior

File: relbench/data/table.py [hunk 1]
```diff
@@ -104,6 +104,15 @@ def upto(self, time_stamp: pd.Timestamp) -> Self:
         if self.time_col is None:
             return self
 
+        is_naive_timestamp = time_stamp.tz is None
+        is_naive_time_col = self.df[self.time_col].dt.tz is None
+        if not is_naive_timestamp == is_naive_time_col:
```
Comment (weihua916) [inline relbench/data/table.py:109]: !=

File: relbench/data/table.py [hunk 2]
```diff
@@ -104,6 +104,15 @@ def upto(self, time_stamp: pd.Timestamp) -> Self:
...
+            if is_naive_timestamp:
+                time_stamp = time_stamp.tz_localize(str(self.df[self.time_col].dt.tz))
+            else:
+                self.df[self.time_col] = self.df["your_datetime_column"].dt.tz_localize(
+                    str(time_stamp.tz)
+                )
```
Comment (weihua916) [inline relbench/data/table.py:115]: what is "your_datetime_column"

File: relbench/external/utils.py [hunk 3]
```diff
@@ -5,8 +5,10 @@
 def to_unix_time(ser: pd.Series) -> np.ndarray:
     r"""Converts a :class:`pandas.Timestamp` series to UNIX timestamp
     (in seconds)."""
-    assert ser.dtype in [np.dtype("datetime64[s]"), np.dtype("datetime64[ns]")]
+    assert str(ser.dtype).startswith("datetime64[s") or str(ser.dtype).startswith(
+        "datetime64[ns"
+    )
     unix_time = ser.astype(int).values
-    if ser.dtype == np.dtype("datetime64[ns]"):
+    if str(ser.dtype).startswith("datetime64[ns"):
```
Comment (weihua916) [inline relbench/external/utils.py:12]: hacky.

File: relbench/data/table.py [hunk 4]
```diff
@@ -104,6 +104,15 @@ def upto(self, time_stamp: pd.Timestamp) -> Self:
         if self.time_col is None:
             return self
 
+        is_naive_timestamp = time_stamp.tz is None
+        is_naive_time_col = self.df[self.time_col].dt.tz is None
```
Comment (weihua916) [inline relbench/data/table.py:108]: Don't understand the logic. What is native

**Review by weihua916 — CHANGES_REQUESTED** [review]
Can you give me more context why this is necessary?

I feel this is too hacky. Can we modify the dataset table side to handle time difference (e.g., normalize the time zone at pre-processing step)?

File: relbench/data/table.py (hunk 4 above)
Comment (yiweny) [reply relbench/data/table.py:108]: it means there's no timezone. If you don't want the logic to handle the timezones, then I can modify the logic while creating datasets

Comment (weihua916) [reply relbench/data/table.py:108]: I think that's better.

**Pull request closed without merging**